### PR TITLE
Intel fpga symbol mapping

### PR DESCRIPTION
**Pull request by TizianoDeMatteis**
Handles symbol mapping for nested SDFG.

Currently, this was causing problems, as nested SDFGs are inlined into OpenCL Kernel. This was resulting in having the kernel with arguments named as the symbols defined in the parent SDFG, while the body of the nestedSDFG was using the inner symbols (hence compiler error).

This PR creates aliases between outer/inner symbols. 
This is supposed to change if we are going to have SDFG as-a-function also for Intel FPGA.

### CODE REVIEW

File: tests/fpga/symbol_mapping.py [hunk 1]
```diff
@@ -0,0 +1,197 @@
+# Copyright 2019-2020 ETH Zurich and the DaCe authors. All rights reserved.
+
+# The scope of the test is to verify that code nested SDFGs with a unique name is generated only once
+# The nested SDFG compute vector addition on FPGA, with vectorization
```
Comment (definelicht) [inline tests/fpga/symbol_mapping.py:4]: How do you check that it is only generated once? I can't see this in the test

**Review by definelicht — CHANGES_REQUESTED** [review]

**Comment by definelicht** [comment]
Why this behavior was different for Intel FPGA compared to the rest of DaCe in the first place?

File: tests/fpga/symbol_mapping.py (hunk 1 above)
Comment (TizianoDeMatteis) [reply tests/fpga/symbol_mapping.py:4]: My fault, wrong comment

File: tests/fpga/symbol_mapping.py [hunk 2]
```diff
@@ -0,0 +1,196 @@
+# Copyright 2019-2020 ETH Zurich and the DaCe authors. All rights reserved.
+
+# The scope of the test is to verify that nestedSDFG mapped symbols are handled correctly
```
Comment (definelicht) [inline tests/fpga/symbol_mapping.py:3]: Please elaborate a bit more on what should happen, and what was wrong before. You can even link the github issue if there is one (e.g., "see #12345").

**Review by definelicht — CHANGES_REQUESTED** [review]

**Comment by TizianoDeMatteis** [comment]
@definelicht I think it is because nested SDFGs are generated as functions. Then the symbol mapping is "handled" directly in the function signature. Xilinx backend defaults to CPU backend.

File: tests/fpga/symbol_mapping.py (hunk 2 above)
Comment (TizianoDeMatteis) [reply tests/fpga/symbol_mapping.py:3]: Mapping was not handled at all.

**Comment by definelicht** [comment]
> @definelicht I think it is because nested SDFGs are generated as functions. Then the symbol mapping is "handled" directly in the function signature. Xilinx backend defaults to CPU backend.

Is there any way we could unite them by using the (relatively new) `arg.ocltype`?

**Pull request closed without merging**